### ISSUE TITLE
prune and valid/test

### ISSUE THREAD

**Issue by MokkeMeguru** [optuna/optuna#1091]
I want to train some deep learning model which consumes too many time and GPU resource.
So I need your prune method in the trial. But I wonder which I should use **test evaluation value** or **valid evaluation value**. (https://github.com/optuna/optuna/blob/master/examples/mlflow/keras_mlflow.py#L78)

We know test evaluated value is no related with valid evaluation value in training. So, I afraid that if I use the test evaluation value, the pruning method will be something wrong I think.

**Comment by toshihikoyanase** [comment]
> But I wonder which I should use test evaluation value or valid evaluation value.

IMO, the validation values are usually used for hyperparameter tuning, and the test values are never used in any training phases (including pruning) to keep the evaluation unbiased.

As you mentioned, some Optuna examples use `test` datasets, but they may be better to be renamed `valid` datasets. What do you think of it?

**Comment by jinnaiyuu** [comment]
I believe that these datasets should be named validation set and avoid calling them test sets.
It may not be that obvious for beginners that we should tune hyperparameters using validation set, not test set.

Actually when I wrote the [lightgbm tuner example](https://github.com/optuna/optuna/blob/master/examples/lightgbm_tuner_simple.py#L21), I intentionally named it validation set...

**Comment by toshihikoyanase** [comment]
@jinnaiyuu That makes sense. I created an issue (#1107) to fix the dataset names.
Thank you!

**Comment by MokkeMeguru** [comment]
I'm waiting for the progress at #1107

**Comment by toshihikoyanase** [comment]
@MokkeMeguru  #1107 was resolved by #1164 and #1167, and let me close this issue.
Please feel free to re-open it if you have any comments.
